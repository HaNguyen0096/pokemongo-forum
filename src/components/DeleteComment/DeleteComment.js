import React from 'react'
import ThreadContext from '../../contexts/ThreadContext'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import config from '../../config'

export default class DeleteThread extends React.Component {
  static defaultProps ={
    onDeleteThread: () => {},
  }
  static contextType = ThreadContext

  handleClickDelete = e => {
    e.preventDefault()
    const { comment } =this.props
    console.log(this.props)
    const commentId = comment.id
    fetch(`${config.API_ENDPOINT}/comments/${commentId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        window.location.reload();
      })
      .then(() => {
        this.context.deleteComment(commentId)
        this.props.onDeleteComment(commentId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    return (
      <div className='Delete_Comment'>
        <button
          className='Delete_Button'
          type='button'
          onClick={this.handleClickDelete}
        >
         <FontAwesomeIcon icon='trash-alt'/>
         {' '}
          remove
        </button>
      </div>
    )
  }
}

