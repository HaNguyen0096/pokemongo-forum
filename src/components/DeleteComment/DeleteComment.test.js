import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import DeleteComment from './DeleteComment'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const testComment = [
  {
    id: 1,
    content: 'test 1',
    user_id: 1,
    thread_id: 1,
    modified: '2020-02-22T21:28:32.615Z'
  },
]

describe('DeleteComment component', () => {

  it('renders DeleteComment without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <DeleteComment comment={testComment}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for DeleteComment as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <DeleteComment comment={testComment} />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders DeleteComment by default', () => {
    const wrapper = shallow(<DeleteComment comment={testComment} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})