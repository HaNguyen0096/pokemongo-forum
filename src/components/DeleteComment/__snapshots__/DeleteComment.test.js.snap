// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteComment component renders DeleteComment by default 1`] = `
<div
  className="Delete_Comment"
>
  <button
    className="Delete_Button"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
</div>
`;

exports[`DeleteComment component renders the UI for DeleteComment as expected 1`] = `
<div
  className="Delete_Comment"
>
  <button
    className="Delete_Button"
    onClick={[Function]}
    type="button"
  >
     
    remove
  </button>
</div>
`;
